.github/workflows/check-user.yml

```diff
@@ -0,0 +1,50 @@
+name: Check whether the workflow owner can use ARC runners
+
+on:
+  workflow_call:
+    inputs:
+      user_name:
+        required: true
+        type: string
+        description: The name of the workflow owner.
+      curr_branch:
+        required: true
+        type: string
+        description: Current branch.
+
+    outputs:
+      workflow-type:
+        description: Type of runners to use
+        value: ${{ jobs.check-user.outputs.workflow-type }}
+
+jobs:
+  check-user:
+    runs-on: linux.4xlarge
+    outputs:
+      workflow-type: ${{ steps.set-condition.outputs.workflow-type }}
+    steps:
+      - name: Check if user is in arc_users.yml
+        id: set-condition
+        run: |
+          curr_branch="${{ inputs.curr_branch }}"
+
+          if [[ "$curr_branch" == "nightly" || "$curr_branch" == "main" || "$curr_branch" == "release/"* || "$curr_branch" == "landchecks/"* ]]; then
+            echo "workflow-type=label" >> $GITHUB_OUTPUT
+          else
+            REPO_OWNER="pytorch"
+            REPO_NAME="test-infra"
+            ISSUE_NUMBER="5132"
+
+            COMMENTS=$(curl https://api.github.com/repos/$REPO_OWNER/$REPO_NAME/issues/$ISSUE_NUMBER/comments)
```

Comment:
See how you can modify this curl command to use `secrets.GITHUB_TOKEN` as an authentication mechanism.  

Github heavily rate limits requests that lack authentication (on a per-ip address basis) and without that auth there's a good chance we'll be affected

Reply:
Ohhh, this is a very good input!